### PR TITLE
[6.x] Adds missing options for PhpRedis - Part 2

### PR DESCRIPTION
**Pull request by danijelk**
In extension of  #31182 we managed to find the issue with serializer tests failing once it was truly enabled. Turns out when using rawCommand in redis that it indeed does send commands in raw format and skips over the native serialiser.

So in order to utilise the serializer, we need to pass data through the native ->command() method.

And since special redis options (XX, NX...) need to be sent as an array, we need to cleanup parameters before passing data to command() method as per phpredis definitions.

We also added a new test for compression option (if compiled), which finally enabled native compression in redis.

### CODE REVIEW

File: src/Illuminate/Redis/Connections/PhpRedisConnection.php [hunk 1]
```diff
@@ -424,6 +440,28 @@ public function flushdb()
      */
     public function executeRaw(array $parameters)
     {
+        if ($this->client->getOption(Redis::OPT_SERIALIZER)) {
+            $method = array_shift($parameters);
+
+            $options = [];
+            $args = [];
+            while (!empty($parameters)) {
```
Comment (driesvints) [inline src/Illuminate/Redis/Connections/PhpRedisConnection.php]:
```suggestion

            while (! empty($parameters)) {
```

File: src/Illuminate/Redis/Connections/PhpRedisConnection.php [hunk 2]
```diff
@@ -424,6 +440,28 @@ public function flushdb()
...
+                if((in_array($parameters[0], ['NX', 'XX', 'CH', 'INCR']))){
+                    $args[] = array_shift($parameters);
+                    continue;
```
Comment (driesvints) [inline src/Illuminate/Redis/Connections/PhpRedisConnection.php]:
```suggestion

                    continue;
```

**Review by driesvints — DISMISSED** [review]

File: src/Illuminate/Redis/Connections/PhpRedisConnection.php [hunk 3]
```diff
@@ -301,6 +301,22 @@ public function zunionstore($output, $keys, $options = [])
         ]);
     }
 
+    /**
+     * Returns the score of member in the sorted set at key
+     *
+     * @param $key
+     * @param $member
```
[the remark this answers is not in the document]
Comment (danijelk) [reply src/Illuminate/Redis/Connections/PhpRedisConnection.php]: indeed forgot return types, and dot from styleci

**Comment by taylorotwell** [comment]
It's unclear what is happening in the code. Extract a method or two so it's more understandable.

**Pull request closed without merging**